presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/IndexInserter.java

```diff
@@ -152,7 +152,19 @@ public void insert(long shardId, UUID shardUuid, OptionalInt bucketNumber, Set<I
     public void execute()
             throws SQLException
     {
-        statement.executeBatch();
+        Connection connection = statement.getConnection();
+        boolean autoCommit = connection.getAutoCommit();
+
+        try {
+            // Turn off the auto commit
+            connection.setAutoCommit(false);
+            statement.executeBatch();
+            connection.commit();
+        }
```

Comment:
I will add it :) 

I didn't do so in the last PR because I rely on the auto close feature to rollback the transaction when exception happens (that's what we did in the previous logic). Catching SQL exceptions and calling ``rollback()`` explicitly does make it clearer. Also we still need to re-throw the SQLException because it need to be handled in the upper layer.